File: 1064-smallest-integer-divisible-by-k/1064-smallest-integer-divisible-by-k.cpp
class Solution {
public:
    int smallestRepunitDivByK(int k) {
        if(k%2==0 || k%5==0){
            return -1;
        }
        int c=0;
        int num=1;
        while(k){
            num%=k;
            c++;
            if(num==0){
                return c;
            }
            num=num*10+1;
        }
        return -1;
        
    }
};

File: 1064-smallest-integer-divisible-by-k/1064-smallest-integer-divisible-by-k.py
class Solution:
    def smallestRepunitDivByK(self, k: int) -> int:
        if(k%2==0 or k%5==0):
            return -1
        num=0
        cou=0
        while k:
            num=num*10+1
            cou+=1
            if(num%k==0):
                return cou
        return -1
        